internal/pkg/template/diff/diff.go

```diff
@@ -88,7 +89,80 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	toSeq, fromSeq := make([]yaml.Node, len(to.Content)), make([]yaml.Node, len(from.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	seen, err := compare(fromSeq, toSeq)
+	if err != nil {
+		return nil, err
+	}
+	var index int
+	children := make(map[string]*Node)
+	for _, occurrence := range seen {
+		if occurrence.inTo == occurrence.inFrom {
+			continue
+		}
+		for i := 0; i < occurrence.inFrom-occurrence.inTo; i++ {
+			children[strconv.Itoa(index)] = &Node{
+				key:      strconv.Itoa(index),
+				oldValue: &(occurrence.node),
+			}
+			index++
+		}
+		for i := 0; i < occurrence.inTo-occurrence.inFrom; i++ {
+			children[strconv.Itoa(index)] = &Node{
+				key:      strconv.Itoa(index),
+				newValue: &(occurrence.node),
+			}
+			index++
+		}
+	}
+	return children, nil
```

Comment:
In Danny's UX: https://github.com/aws/copilot-cli/issues/4259 (which is inspired by Terraform https://www.hashicorp.com/blog/terraform-0-14-adds-a-new-concise-diff-format-to-terraform-plans) there is also some metadata around the number of elements that do not change:
```
# (2 unchanged elements hidden)
```

I wonder how we can capture that info 💭  Perhaps it warrants being a field in `Node`